vernac/vernacentries.ml

```diff
@@ -1482,6 +1482,24 @@ let vernac_declare_instance ~atts id bl inst pri =
   in
   Classes.declare_new_instance ~program_mode:program ~locality ~poly id bl inst pri
 
+let vernac_context ~atts ctx =
+  let sec = Lib.sections_are_opened () in
+  let program_mode, poly = Attributes.(parse (Notations.(program ++ polymorphic))) atts in
+  if Dumpglob.dump () then begin
+    let l = List.map (function
+        | Constrexpr.CLocalAssum (l, _, _, _) ->
+           let ty = if sec then "var "else "ax" in List.map (fun n -> ty, n) l
+        | Constrexpr.CLocalDef (n, _, _, _) ->
+           let ty = if sec then "var "else "def" in [ty, n]
+        | Constrexpr.CLocalPattern _ -> [])
+      ctx in
+    List.iter (function
+        | ty, {CAst.v = Names.Name.Anonymous; _} -> ()
+        | ty, {CAst.v = Names.Name.Name id; loc} ->
+           Dumpglob.dump_definition (CAst.make ?loc id) sec ty)
+      (List.flatten l) end;
+  ComAssumption.do_context ~program_mode ~poly ctx
```

Comment:
We should probably consider (in further PRs) to move the dumpglobs to the com* files. vernacentires should probably be reserved for attributes parsing and dispatching to the various com* files, its already pretty bulky. @SkySkimmer might have a stronger opinion than me here
Meanwhile, regarding the current PR maybe we can just drop the commit " Move dumpglob of Context to vernacentries.ml, like it is already done for Variables." and do the reverse for variables?

Reply:
Done, I'll merge by the end of the week if there are no more comments.